ui/lib/utils/DI.md

```diff
@@ -0,0 +1,52 @@
+# Dependency Injection (DI) Documentation
+
+This document describes how the simple DI function works in React.
+
+And we can organize the code in a more SOLID way through the SOA.
+
+## Usage
+
+Define service & token:
+
+```tsx
+// MyService.ts
+import { useToken } from '@lib/utils/di'
+
+// service impl
+export const useMyService = () => {
+  // ...
+}
+
+// service token
+export const MyService = useToken(useMyService)
```

Comment:
What is a "token"?

Reply:
Yes, just a lookup token associated with a dependency provider.
And it provide better type experience in ts. I should redesign `useToken` so that users can easily use ts interface, factory or static value.